regexp.h

```diff
@@ -152,17 +160,10 @@ typedef struct regexp {
     /* Information about the match that isn't often used */
 
     SSize_t maxlen;  /* minimum possible number of chars in string to match */
+    char *subbeg;       /* saved or original string so \digit works forever. */
```

Comment:
This appears to suggest subbeg is now "information about the match that isn't often used". If it works, maybe it would be better to move qr_anoncv up before subbeg's original location.

Reply:
Or mv subbeg to just after suboffset, and perhaps qr_anoncv_up to between that and maxlen.  

I missed that the commit changed groupings, and I think it better to retain those, with the importance lessening as we go down the struct

Reply:
Moving subbeg to the other side of that comment was something I wasn't sure about from the human-readability perspective.  Had toyed with moving qr_anoncv too.

Would you like me to compile either/both of your suggestions and post the pahole output? (I may be offline during this week, so apologies for any delay.) If so, please could you also confirm which of the following moves from the commit should also be rolled in:
- nparens
- intflags
- lastcloseparen
- pre_prefix & compflags

Reply:
I think your PR is fine in regard to all other changes in it.  So, I'd like you to try both our suggestions to see what happens

Reply:
I tried both your suggestions and, after looking again at the four moves mentioned above, not doing any additional moves at all. The three options look the same w.r.t. alignment holes. There was no need to move subbeg in the first place; probably I had moved it when trying different reorganizations and missed it when cleaning up for the PR. Sorry about that! Will push an updated PR shortly. pahole output below FYI.

No subbeg or other later moves:

```
struct regexp {
	HV *                       xmg_stash;            /*     0     8 */
	union _xmgu        xmg_u;                        /*     8     8 */
	STRLEN                     xpv_cur;              /*    16     8 */
	union {
		STRLEN             xpvlenu_len;          /*    24     8 */
		struct regexp *    xpvlenu_rx;           /*    24     8 */
	} xpv_len_u;                                     /*    24     8 */
	const struct regexp_engine  * engine;            /*    32     8 */
	REGEXP *                   mother_re;            /*    40     8 */
	HV *                       paren_names;          /*    48     8 */
	U32                        extflags;             /*    56     4 */
	U32                        nparens;              /*    60     4 */
	/* --- cacheline 1 boundary (64 bytes) --- */
	ssize_t                    minlen;               /*    64     8 */
	ssize_t                    minlenret;            /*    72     8 */
	STRLEN                     gofs;                 /*    80     8 */
	struct reg_substr_data *   substrs;              /*    88     8 */
	void *                     pprivate;             /*    96     8 */
	U32                        intflags;             /*   104     4 */
	U32                        lastparen;            /*   108     4 */
	regexp_paren_pair *        offs;                 /*   112     8 */
	char * *                   recurse_locinput;     /*   120     8 */
	/* --- cacheline 2 boundary (128 bytes) --- */
	U32                        lastcloseparen;       /*   128     4 */
	U32                        pre_prefix:4;         /*   132: 0  4 */
	U32                        compflags:9;          /*   132: 4  4 */

	/* XXX 19 bits hole, try to pack */

	char *                     subbeg;               /*   136     8 */
	SV *                       saved_copy;           /*   144     8 */
	ssize_t                    sublen;               /*   152     8 */
	ssize_t                    suboffset;            /*   160     8 */
	ssize_t                    subcoffset;           /*   168     8 */
	ssize_t                    maxlen;               /*   176     8 */
	CV *                       qr_anoncv;            /*   184     8 */

	/* size: 192, cachelines: 3, members: 28 */
	/* sum members: 188 */
	/* sum bitfield members: 13 bits, bit holes: 1, sum bit holes: 19 bits */
};
```
hvds's idea, move qr_anoncv up before subbeg's original location:
```
struct regexp {
	HV *                       xmg_stash;            /*     0     8 */
	union _xmgu        xmg_u;                        /*     8     8 */
	STRLEN                     xpv_cur;              /*    16     8 */
	union {
		STRLEN             xpvlenu_len;          /*    24     8 */
		struct regexp *    xpvlenu_rx;           /*    24     8 */
	} xpv_len_u;                                     /*    24     8 */
	const struct regexp_engine  * engine;            /*    32     8 */
	REGEXP *                   mother_re;            /*    40     8 */
	HV *                       paren_names;          /*    48     8 */
	U32                        extflags;             /*    56     4 */
	U32                        nparens;              /*    60     4 */
	/* --- cacheline 1 boundary (64 bytes) --- */
	ssize_t                    minlen;               /*    64     8 */
	ssize_t                    minlenret;            /*    72     8 */
	STRLEN                     gofs;                 /*    80     8 */
	struct reg_substr_data *   substrs;              /*    88     8 */
	void *                     pprivate;             /*    96     8 */
	U32                        intflags;             /*   104     4 */
	U32                        lastparen;            /*   108     4 */
	regexp_paren_pair *        offs;                 /*   112     8 */
	char * *                   recurse_locinput;     /*   120     8 */
	/* --- cacheline 2 boundary (128 bytes) --- */
	U32                        lastcloseparen;       /*   128     4 */
	U32                        pre_prefix:4;         /*   132: 0  4 */
	U32                        compflags:9;          /*   132: 4  4 */

	/* XXX 19 bits hole, try to pack */

	CV *                       qr_anoncv;            /*   136     8 */
	char *                     subbeg;               /*   144     8 */
	SV *                       saved_copy;           /*   152     8 */
	ssize_t                    sublen;               /*   160     8 */
	ssize_t                    suboffset;            /*   168     8 */
	ssize_t                    subcoffset;           /*   176     8 */
	ssize_t                    maxlen;               /*   184     8 */

	/* size: 192, cachelines: 3, members: 28 */
	/* sum members: 188 */
	/* sum bitfield members: 13 bits, bit holes: 1, sum bit holes: 19 bits */
};
```
khw's idea, move subbeg and qr_anoncv
```
struct regexp {
	HV *                       xmg_stash;            /*     0     8 */
	union _xmgu        xmg_u;                        /*     8     8 */
	STRLEN                     xpv_cur;              /*    16     8 */
	union {
		STRLEN             xpvlenu_len;          /*    24     8 */
		struct regexp *    xpvlenu_rx;           /*    24     8 */
	} xpv_len_u;                                     /*    24     8 */
	const struct regexp_engine  * engine;            /*    32     8 */
	REGEXP *                   mother_re;            /*    40     8 */
	HV *                       paren_names;          /*    48     8 */
	U32                        extflags;             /*    56     4 */
	U32                        nparens;              /*    60     4 */
	/* --- cacheline 1 boundary (64 bytes) --- */
	ssize_t                    minlen;               /*    64     8 */
	ssize_t                    minlenret;            /*    72     8 */
	STRLEN                     gofs;                 /*    80     8 */
	struct reg_substr_data *   substrs;              /*    88     8 */
	void *                     pprivate;             /*    96     8 */
	U32                        intflags;             /*   104     4 */
	U32                        lastparen;            /*   108     4 */
	regexp_paren_pair *        offs;                 /*   112     8 */
	char * *                   recurse_locinput;     /*   120     8 */
	/* --- cacheline 2 boundary (128 bytes) --- */
	U32                        lastcloseparen;       /*   128     4 */
	U32                        pre_prefix:4;         /*   132: 0  4 */
	U32                        compflags:9;          /*   132: 4  4 */

	/* XXX 19 bits hole, try to pack */

	SV *                       saved_copy;           /*   136     8 */
	ssize_t                    sublen;               /*   144     8 */
	ssize_t                    suboffset;            /*   152     8 */
	ssize_t                    subcoffset;           /*   160     8 */
	char *                     subbeg;               /*   168     8 */
	CV *                       qr_anoncv;            /*   176     8 */
	ssize_t                    maxlen;               /*   184     8 */

	/* size: 192, cachelines: 3, members: 28 */
	/* sum members: 188 */
	/* sum bitfield members: 13 bits, bit holes: 1, sum bit holes: 19 bits */
};
```